int32 len
MoveCmd[] list
